plugins/versionpress/tests/Automation/WpAutomation.php

```diff
@@ -135,11 +135,15 @@ public function uninstallVersionPress()
      *
      * @see wp_insert_post()
      * @param array $post (as wp_insert_post)
+     * @param bool $waitOneSecond Wait to ensure that there are no testing posts published at the same time.
+     *                            Currently used only for End2End tests which would fail without it, see #1420.    
      * @return int
      */
-    public function createPost(array $post)
+    public function createPost(array $post, $waitOneSecond = false)
```

Comment:
Though it's a step in the right direction, this is still possible:

```php
$post1["post_date"] = "2011-11-11 11:11:11";
$post2["post_date"] = "2011-11-11 11:11:11";

$wpAutomation->createPost($post1, true);
$wpAutomation->createPost($post2, true);

// date collision not really prevented
```  

I think the method could look something like this:

```php
public function createPost(array $post, $managePostDateAutomatically = false)
{
    if ($managePostDateAutomatically) {
        if (isset($post["post_date"])) {
            throw new Exception('post_date must not be set when managing post date automatically');
        }
        sleep(1);
    }
    // ...
}
```

The usage of `sleep` becomes an implementation detail that could change in the future, the main thing is that the method guarantees certain behavior.